### PR TITLE
issue/#7 create/edit CONTRIBUTING

### LINKED ISSUE #7: Edit CONTRIBUTING.md
- [x] Move over any useful astro instructions from README.md
- [x] Add branch naming conventions / git flow / branch rules
- [x] Add conventional commits tl;dr and link
- [x] Add testing expectations
- [x] Add process for developing/deploying/documenting microservices (separate repos - linked)
- [x] Add CI flow description 
- [x] Add pre-commit setup description

### PR DESCRIPTION
**Pull request by CKCarr**
Followed checklist based upon issue #7

addional Changes include:

This would relate to issue #1 adding a pre-commit
created `pre-commit-config.yaml` setup file to standardize code quality checks.

This was not directly related to any other issues
created Code of Conduct - linked in Contributing for easy access. 

Closes #7

Please review and suggest any further insights, improvements or modifications needed.
Is this comprehensive and easy to follow? How could I improve?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+| :------------------------ | :----------------------------------------------- |
+| `npm install`             | Installs dependencies                            |
+| `npm run dev`             | Starts local dev server at `localhost:4321`      |
+| `npm run build`           | Build your production site to `./dist/`          |
+| `npm run preview`         | Preview your build locally, before deploying     |
+| `npm run astro ...`       | Run CLI commands like `astro add`, `astro check` |
+| `npm run astro -- --help` | Get help using the Astro CLI                     |
+
+## Reporting Bugs
+Report bugs by opening a new issue. Please include a clear description, as much relevant information as possible, and a code sample or an executable test case demonstrating the expected behavior that is not occurring.
```
Comment (helmturner) [inline CONTRIBUTING.md:198]:
```suggestion
Report bugs by opening a new issue. Please include a clear description, as much relevant information as possible, and, most importantly, a [minimum reproducible example (instructions)](https://stackoverflow.com/help/minimal-reproducible-example).
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
```
Comment (helmturner) [inline .pre-commit-config.yaml:8]: Prettier has a pre-commit hook, too. Here's the config suggestion:

```suggestion
    -   id: check-added-large-files
-   repo: https://github.com/pre-commit/mirrors-prettier
    name: Prettier
    rev: "3.2.5"
    hooks:
    -   id: prettier
        additional_dependencies:
        -   prettier@3.2.5
        -   prettier-plugin-astro@0.13.0
    	-   prettier-plugin-tailwindcss@0.5.11
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
```
Comment (helmturner) [inline .pre-commit-config.yaml:6]: Prettier handles both of these, so these can be removed in favor of the Prettier pre-commit hook.
```suggestion
```

**Review by helmturner — CHANGES_REQUESTED** [review]
Great work! I've just added some suggestions for the pre-commit hook and a link to a how-to for minimum reproducible examples

File: .pre-commit-config.yaml (hunk 3 above)
Comment (zenlex) [reply .pre-commit-config.yaml:6]: @CKCarr I'm good with this with @helmturner 's requested changes. Mark it ready for review when those are implemented and either one of us can approve it and get it merged 🎉 :shipit:

**Review by helmturner — APPROVED** [review]

**Comment by helmturner** [comment]
I'm getting "PR not mergable". I'm on mobile, so not able to see why easily... if necessary, can you please rebase on top of dev?

**Pull request merged**